manifests/charts/base/templates/crds.yaml

```diff
@@ -1,13 +1,21 @@
+# TODO enableCRDTemplates is now defaulted to true as we
+# want to always self-manage CRD upgrades via plain templates,
+# so we should remove this flag after a few releases
 {{- if .Values.base.enableCRDTemplates }}
-{{- if .Values.base.excludedCRDs }}
-{{- range $crd := .Files.Get "crds/crd-all.gen.yaml"|splitList "\n---\n"}}
+{{- $replacement := include "istio.labels" . | fromYaml}}
+{{- range $crd := .Files.Get "files/crd-all.gen.yaml"|splitList "\n---\n"}}
 {{- $name := (index ($crd |fromYaml) "metadata" "name") }}
 {{- if not (has $name $.Values.base.excludedCRDs)}}
-{{$crd}}
+{{- $asDict := ($crd | fromYaml) }}
+# If we are templating these CRDs, we want to wipe out the "static"/legacy
+# labels and replace them with the standard templated istio ones.
+# This allows the continued use of `kubectl apply -f crd-all.gen.yaml`
+# without any templating+the old labels, if desired.
```

Comment:
hmm, wonder if this will break any usage that may depend on the old labels...? `heritage: Tiller` is certainly a silly one but maybe the rest less so.

Note we can also change the default ones coming into crd-all.gen.yaml to be more reasonable. WE will still want this since some of the labels are dynamic, but like managed-by, part-of are static I think. or just remove them. Since I think some people will still do `kubectl apply ...` for various reasons

Reply:
That's why I left `enableCRDTemplates` in - if that's false you will get (ultimately) the exact same rendered output as before this change.

You can `helm install` and get the new/correct labels via templating, or you can `kubectl apply -f` and you get the static autogen'd labels of yore, if for some reason you want them.

Later on we can maybe remove this flag _and_ remove those labels from the autogen but in an excess of caution I'm just leaving the back compat flag here.

Reply:
ack, makes sense, I just misunderstood which aspects of compatibility we were retaining

Reply:
https://github.com/istio/istio/issues/53361 just to keep track.